### PR TITLE
KAFKA-8430: unit test to make sure null `group.id` and valid `group.instance.id` are valid combo

### PR DESCRIPTION
**Pull request by abbccdda**
as title suggests, this unit test is just a double check. No need to push in 2.3

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by abbccdda** [comment]
@mjsax @guozhangwang FYI

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java [hunk 1]
```diff
@@ -210,6 +210,18 @@ public void testInvalidSocketReceiveBufferSize() {
         new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
     }
 
+    @Test
+    public void testInitSuccessWithDefaultNullGroupIdAndValidGroupInstanceId() {
```
Comment (mjsax) [inline clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java]: nit: `shouldIgnoreGroupInstanceIdForEmptyGroupId`

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java [hunk 2]
```diff
@@ -210,6 +210,18 @@ public void testInvalidSocketReceiveBufferSize() {
...
+        Map<String, Object> config = new HashMap<>();
+        config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        config.put(ConsumerConfig.SEND_BUFFER_CONFIG, Selectable.USE_DEFAULT_BUFFER_SIZE);
+        config.put(ConsumerConfig.RECEIVE_BUFFER_CONFIG, Selectable.USE_DEFAULT_BUFFER_SIZE);
```
Comment (mjsax) [inline clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java]: Why do we need to set `SEND_BUFFER_CONFIG` and `RECEIVE_BUFFER_CONFIG` ?

Comment (abbccdda) [reply clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java]: No we don't lol

**Review by guozhangwang — COMMENTED** [review]
LGTM. Waiting for green builds.

**Comment by abbccdda** [comment]
Retest this please

**Comment by guozhangwang** [comment]
Cherry-picked to 2.3 as well.

**Pull request merged**